### PR TITLE
Feature/add styles to mailer and add tests

### PR DESCRIPTION
**Pull request by bogdanrymar**
Add some styles for AnswerMailer, add tests for AnswersController and AnswerMailer

### CODE REVIEW

File: app/controllers/admins/answers_controller.rb [hunk 1]
```diff
@@ -10,7 +10,8 @@ def create
         AnswerMailer.with(answer: @answer, question: @question, admin: current_user).question_answered.deliver_later
         redirect_to admins_question_path(@question), notice: 'Answer created!'
       else
-        redirect_to admins_question_path(@question), notice: 'Error, something goes wrong'
+        @answers = @question.answers.all
```
Comment (mmrshk) [inline app/controllers/admins/answers_controller.rb:13]: `@question.answers`

File: app/views/admins/questions/show.html.erb [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
+<%= @question.errors.full_messages %>
 <p><%= @question.body %></p>
 <p><%= @question.user_name %></p>
 <p><%= @question.user_email %></p>
 
 <h1>Create Answer</h1>
 <%= simple_form_for ([:admins, @question, @answer]) do |form| %>
   <%= form.input :body, label: 'Body' %>
-  <%= form.button :submit, class: "btn-primary"%>
+  <%= form.button :submit%>
```
Comment (mmrshk) [inline app/views/admins/questions/show.html.erb:10]: `<%= form.button :submit %>`

File: spec/mailers/answer_mailer_spec.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe AnswerMailer, type: :mailer do
+  before do
+    allow(Rails.application.credentials.google_account).to receive(:[]).with(:SMTP_USERNAME).and_return(sender_email)
+  end
```
Comment (mmrshk) [inline spec/mailers/answer_mailer_spec.rb:8]: empty line

Comment (mmrshk) [reply spec/mailers/answer_mailer_spec.rb:8]: set `SMTP_USERNAME` for test env

File: app/views/admins/questions/show.html.erb [hunk 4]
```diff
@@ -1,13 +1,13 @@
 <h1><%= @question.title %></h1>
-
+<%= @question.errors.full_messages %>
```
Comment (mmrshk) [inline app/views/admins/questions/show.html.erb:2]: remove

File: spec/mailers/answer_mailer_spec.rb [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  describe 'question_answered' do
+    let(:mail) { AnswerMailer.with(answer: answer, question: question, admin: answered_by).question_answered }
```
Comment (mmrshk) [inline spec/mailers/answer_mailer_spec.rb:10]: `subject { described_class.with(answer: answer, question: question, admin: answered_by).question_answered }`

File: spec/requests/admins/answers_spec.rb [hunk 6]
```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'Answers', type: :request do
+  let!(:question) { create(:question, answers: [answer]) }
+  let!(:answer) { create(:answer) }
+
```
Comment (mmrshk) [inline spec/requests/admins/answers_spec.rb:8]:
```
let!(:question) { create(:question, :with_answer)
let!(:answer) { question.answers.first )
```

File: spec/requests/admins/answers_spec.rb [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+        expect(response).to redirect_to admins_question_path(question, locale: I18n.locale)
+        follow_redirect!
+        expect(response).to have_http_status(:success)
+        expect(response).to render_template(:show)
+        expect(response.body).to include('Answer created!')
+      end
+    end
+    context 'with invalid parameters' do
+      let(:invalid_answer) { attributes_for(:answer, :invalid_answer) }
+      let(:invalid_params) { attributes_for(:answer, :invalid_answer) }
```
Comment (mmrshk) [inline spec/requests/admins/answers_spec.rb:54]: empty line

Comment (mmrshk) [reply spec/requests/admins/answers_spec.rb:54]: remove one of this

**Review by mmrshk — CHANGES_REQUESTED** [review]
Re-look your PR.

File: app/controllers/admins/answers_controller.rb (hunk 1 above)
Comment (bogdanrymar) [reply app/controllers/admins/answers_controller.rb:13]: It gives me error

Comment (mmrshk) [reply app/controllers/admins/answers_controller.rb:13]: which?

File: spec/mailers/answer_mailer_spec.rb [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+    let(:answered_by) { create(:user, :admin) }
+
+    it 'renders the headers' do
+      expect(subject.subject).to eq('Question answered')
+      expect(subject.to).to eq([question.user_email])
+      expect(subject.from).to eq(['innovation.forpost@gmail.com'])
+    end
+
+    it 'contains necessary info' do
+      expect(subject.body.encoded).to match question.title
```
Comment (mmrshk) [inline spec/mailers/answer_mailer_spec.rb:19]: decide which style you prefer more and use it (with braces or without) 

`expect(subject.body.encoded).to match(question.title)`

I guess we use more braces style. Please, don't mix.

File: spec/requests/admins/answers_spec.rb [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+      it 'create answer with valid params' do
+        post admins_question_answers_path(question), params: { answer: params }
+        expect(response).to have_http_status(:redirect)
+        expect(response).to redirect_to admins_question_path(question, locale: I18n.locale)
+        follow_redirect!
+        expect(response).to have_http_status(:success)
+        expect(response).to render_template(:show)
+        expect(response.body).to include('Answer created!')
+      end
+    end
```
Comment (mmrshk) [inline spec/requests/admins/answers_spec.rb:50]: empty line between contexts!

File: spec/requests/admins/answers_spec.rb [hunk 10]
```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'Answers', type: :request do
+  let!(:question) { create(:question, :question_with_answer) }
+  let!(:answer) { question.answers.first }
+
```
Comment (mmrshk) [inline spec/requests/admins/answers_spec.rb:8]: redundant line

**Pull request merged**